src/modules/Host.rb

```diff
@@ -289,14 +292,22 @@ def StaticIPs
       static_ips
     end
 
-    # if we have a static address,
-    # make sure /etc/hosts resolves it to our, bnc#664929
+    # Configure system to resolve static ips without hostname to system wide hostname
+    #
+    # It is expected to be used during installation only. If user configures static
+    # ips during installation and do not assign them particular hostname, then such
+    # ips are configuret to resolve to the system wide hostname (see Hostname module,
+    # /etc/HOSTNAME)
+    #
+    # Originally implemented as a fix for bnc#664929, later extended for bnc#1039532
     def ResolveHostnameToStaticIPs
-      static_ips = StaticIPs()
+      # reject those static ips which have particular hostnames already configured
+      static_ips = StaticIPs().reject { |sip| @hosts.include_ip?(sip) }
```

Comment:
so looks like variable name is no longer descriptive as it is not static_ips, but static ips without hostname?